src/Controls/src/Build.Tasks/NodeILExtensions.cs

```diff
@@ -161,25 +161,27 @@ static T TryFormat<T>(Func<string, T> func, IXmlLineInfo lineInfo, string str)
 			}
 		}
 
-		static Dictionary<TypeReference, Type> CompiledTypeConverters;
+		static Dictionary<TypeReference, Type> KnownCompiledTypeConverters;
 
 		public static IEnumerable<Instruction> PushConvertedValue(this ValueNode node, ILContext context,
 			TypeReference targetTypeRef, TypeReference typeConverter, IEnumerable<Instruction> pushServiceProvider,
 			bool boxValueTypes, bool unboxValueTypes)
 		{
 			var module = context.Body.Method.Module;
-			if (CompiledTypeConverters == null)
+			if (KnownCompiledTypeConverters == null)
 			{
-				CompiledTypeConverters = new Dictionary<TypeReference, Type>();
-				CompiledTypeConverters.Add(module.ImportReference(("Microsoft.Maui", "Microsoft.Maui", "Thickness")), typeof(ThicknessTypeConverter));
-				CompiledTypeConverters.Add(module.ImportReference(("Microsoft.Maui.Graphics", "Microsoft.Maui.Graphics", "Color")), typeof(ColorTypeConverter));
+				KnownCompiledTypeConverters = new Dictionary<TypeReference, Type>
```

Comment:
Is it possible to use the `TypeRefComparer.Default` here? Then the `DictionaryExtensions.TryGetValue` should not be needed.

```suggestion
				KnownCompiledTypeConverters = new Dictionary<TypeReference, Type>(TypeRefComparer.Default)
```

Reply:
I actually looked for that API, knowing that it existed, and was unable to find it, hence the extension method. But suddenly here it is. What cursed magic is this, again ??

Reply:
fixed